models/actions/schedule.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package actions
+
+import (
+	"context"
+
+	"code.gitea.io/gitea/models/db"
+	repo_model "code.gitea.io/gitea/models/repo"
+	user_model "code.gitea.io/gitea/models/user"
+	"code.gitea.io/gitea/modules/timeutil"
+	webhook_module "code.gitea.io/gitea/modules/webhook"
+)
+
+// ActionSchedule represents a schedule of a workflow file
+type ActionSchedule struct {
+	ID            int64
+	Title         string
+	Specs         []string
+	EntryIDs      []int                  `xorm:"entry_ids"`
+	RepoID        int64                  `xorm:"index"`
+	Repo          *repo_model.Repository `xorm:"-"`
+	OwnerID       int64                  `xorm:"index"`
+	WorkflowID    string                 `xorm:"index"` // the name of workflow file
+	TriggerUserID int64
+	TriggerUser   *user_model.User `xorm:"-"`
+	Ref           string
+	CommitSHA     string
+	Event         webhook_module.HookEventType
+	EventPayload  string `xorm:"LONGTEXT"`
```

Comment:
According to the [doc](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule):
- Event should be always `schedule`.
- CommitSHA should be the last commit on default branch.
- Ref should be always the default branch.

So they should not be stored in DB.

And I didn't check other fields, maybe I'm missing something.

Reply:
But they should still be stored because main branch maybe changed and branch maybe move to a new commit.